scripts/build-preflash.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Copyright (c) 2025 Tenstorrent AI ULC
+# SPDX-License-Identifier: Apache-2.0
+
+# Helper script to build preflash bootfs image for assembly
+set -e
+
+# handle temp directory for the build
+if [ -z "$BUNDLE_TEMP_PREFIX" ]; then
+  TEMP_DIR="$(mktemp -d)/"
+  echo "Created TEMP_DIR"
+else
+  echo "Using existing BUNDLE_TEMP_PREFIX: $BUNDLE_TEMP_PREFIX"
+  TEMP_DIR="$BUNDLE_TEMP_PREFIX"
+fi
+
+cleanup() {
+    if [ -z "$BUNDLE_TEMP_PREFIX" ] && [ -d "$TEMP_DIR" ]; then
+        rm -Rf "$TEMP_DIR"
+        echo "Removed TEMP_DIR"
+    fi
+}
+trap cleanup EXIT
+
+# Get the script directory and base paths
+SCRIPT_DIR="$(dirname "$0")"
+TTZP_BASE="$(dirname "$SCRIPT_DIR")"
+PATH="$SCRIPT_DIR:$PATH"
+
+# board revision is agnostic, we use p100 for preflash
+BOARD_REV="p100"
+BUILD_DIR="${TEMP_DIR}"
+
+# firmware version info
+MAJOR="$(grep "^VERSION_MAJOR" "$TTZP_BASE/VERSION" | awk '{print $3}')"
+MINOR="$(grep "^VERSION_MINOR" "$TTZP_BASE/VERSION" | awk '{print $3}')"
+PATCH="$(grep "^PATCHLEVEL" "$TTZP_BASE/VERSION" | awk '{print $3}')"
+EXTRAVERSION="$(grep "^EXTRAVERSION" "$TTZP_BASE/VERSION" | awk '{print $3}')"
+
+if [ -z "$MAJOR" ] || [ -z "$MINOR" ] || [ -z "$PATCH" ]; then
+  echo "required version info missing: MAJOR=$MAJOR MINOR=$MINOR PATCH=$PATCH"
+  exit 1
+fi
+
```

Comment:
Without this, you'll get e.g. `preflash-18.6.0rc3.ihex` instead of `preflash-18.6.0-rc3.ihex`

```suggestion

if [ -n "$EXTRAVERSION" ]; then
  EXTRAVERSION="-$EXTRAVERSION"
  EXTRAVERSION_NUMBER="$(echo "$EXTRAVERSION" | sed 's/[^0-9]*//g')"
fi
```